src/json-ld.net/Core/DocumentLoader.cs

```diff
@@ -6,31 +6,73 @@
 using JsonLD.Util;
 using System.Net;
 using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using System.Runtime.InteropServices;
 
 namespace JsonLD.Core
 {
     public class DocumentLoader
     {
+        const int MAX_REDIRECTS = 20;
+
+        /// <summary>An HTTP Accept header that prefers JSONLD.</summary>
+        /// <remarks>An HTTP Accept header that prefers JSONLD.</remarks>
+        public const string AcceptHeader = "application/ld+json, application/json;q=0.9, application/javascript;q=0.5, text/javascript;q=0.5, text/plain;q=0.2, */*;q=0.1";
+
+        /// <exception cref="JsonLDNet.Core.JsonLdError"></exception>
+        public RemoteDocument LoadDocument(string url)
+        {
+            return LoadDocumentAsync(url).GetAwaiter().GetResult();
+        }
+
         /// <exception cref="JsonLDNet.Core.JsonLdError"></exception>
-        public virtual RemoteDocument LoadDocument(string url)
+        public async Task<RemoteDocument> LoadDocumentAsync(string url)
         {
-#if !PORTABLE && !IS_CORECLR
             RemoteDocument doc = new RemoteDocument(url, null);
-            HttpWebResponse resp;
 
             try
             {
-                HttpWebRequest req = (HttpWebRequest)HttpWebRequest.Create(url);
-                req.Accept = AcceptHeader;
-                resp = (HttpWebResponse)req.GetResponse();
-                bool isJsonld = resp.Headers[HttpResponseHeader.ContentType] == "application/ld+json";
-                if (!resp.Headers[HttpResponseHeader.ContentType].Contains("json"))
+                HttpResponseMessage httpResponseMessage;
+
+                int redirects = 0;
+                int code;
+                string redirectedUrl = url;
+
+                // Manually follow redirects because .NET Core refuses to auto-follow HTTPS->HTTP redirects.
```

Comment:
nasty. I wonder if we should pull this out into its own class and put some unit tests around it.

Reply:
All for pull it into its own class.

The conformance tests already cover it surprisingly well: https://github.com/linked-data-dotnet/json-ld.net/blob/master/test/json-ld.net.tests/W3C/remote-doc-manifest.jsonld#L43-L127

Reply:
Factored out. Pretty happy with how it turned out.